dbloader/riak_loader.py

```diff
@@ -0,0 +1,170 @@
+        return
+
+    def insert(self, bucket, table=None, custom=None):
+        '''
+        Insert a single key/value
+        '''
+
+        logger.debug(' - Insert key %d', custom)
+        start_time = time.time()
+        if not self.conn:
+            self.get_connection()
+        try:
+            b = self.conn.bucket(bucket)
+            random_text = Loader.big_string(100)
+            value = {"type": "Load Test",
+                     "randString": random_text,
+                     "created": start_time,
+                     "custom": custom,
+                     "concurrency": 1}
+            kv = b.new(key=str(custom), data=value, content_type='application/json')
+            result = kv.store()
+            stored = b.get(kv.key)
+
+        except Exception:
+            logger.exception('Unable to insert an object')
+            return False
+        logger.debug(' - Inserted %s (%s)', result.key, kv.data)
+        logger.debug(' - Saved! (%s)', stored.data)
+        return time.time() - start_time
+
+    def update(self, bucket, table=None, custom=None):
+        '''
+        Update a single key/value
+        '''
+
+        start_time = time.time()
+        if not self.conn:
+            self.get_connection()
+        try:
+            b = self.conn.bucket(bucket)
+            kv = b.get(str(custom))
+            random_text = Loader.big_string(100)
+            value = {"type": "Load Test",
+                     "randString": random_text,
+                     "created": start_time,
+                     "updated": start_time,
+                     "concurrency": 1}
+            kv.data = value
+            result = kv.store()
+
+        except Exception:
+            logger.exception('Unable to update an object')
+            return False
+        return time.time() - start_time
+
+    def delete(self, bucket, table, custom=None):
+        '''
+        Delete a single object
+        '''
+        start_time = time.time()
+        if not self.conn:
+            self.get_connection()
+        try:
+            b = self.conn.bucket(bucket)
+            result = b.delete(str(custom))
+
+        except Exception:
+            logger.exception('Unable to delete key')
+            raise Exception
+        return time.time() - start_time
+
+    def select(self, bucket, table=None, custom=None):
+        '''
+        Select a single object
+        '''
+        start_time = time.time()
+        if not self.conn:
+            self.get_connection()
+        try:
+            b = self.conn.bucket(bucket)
+            result = b.get(str(custom))
+
+        except Exception:
+            logger.exception('Unable to select key')
+            raise Exception
+        return time.time() - start_time
+
+    def insert_some(self, custom=None):
+        '''
+        Load data into a bucket
+        '''
+        if not self.conn:
+            self.get_connection()
+        if not self.ready:
+            self.create_if_not_exists(custom)
```

Comment:
Any reason for checking self.ready and calling create_if_not_exists? We have not assigned a value to self.ready for Riak loader and haven't implemented create_if_not_exists for Riak. What it will be doing in this case would be call parent's create_if_not_exists method which is doing nothing here: https://github.com/phil-hildebrand/dbloader/blob/de87a12036a3d9c7571d3e0263e79d3d2df55ff2/dbloader/__init__.py#L53. So, i believe, if we are not planning on implementing it here, we should just remove the condition.